cmd/fleetctl/mdm.go

```diff
@@ -164,3 +166,144 @@ fleetctl get mdm-command-results --id=%v
 		},
 	}
 }
+
+func mdmLockCommand() *cli.Command {
+	return &cli.Command{
+		Name:  "lock",
+		Usage: "Lock a host when it needs to be returned to your organization.",
+		Flags: []cli.Flag{contextFlag(), debugFlag(), &cli.StringFlag{
+			Name:     "host",
+			Usage:    "The host, specified by identifier, that you want to lock.",
+			Required: true,
+		}},
+		Action: func(c *cli.Context) error {
+			hostIdent := c.String("host")
+
+			if len(hostIdent) == 0 {
+				return errors.New("No host targeted. Please provide --host.")
+			}
+
+			client, err := clientFromCLI(c)
+			if err != nil {
+				return fmt.Errorf("create client: %w", err)
+			}
+
+			host, err := client.HostByIdentifier(hostIdent)
+			if err != nil {
+				var nfe service.NotFoundErr
+				if errors.As(err, &nfe) {
+					return errors.New("The host doesn't exist. Please provide a valid host identifier.")
+				}
+
+				var sce kithttp.StatusCoder
+				if errors.As(err, &sce) {
+					if sce.StatusCode() == http.StatusForbidden {
+						return errors.New("Permission denied. You don't have permission to lock this host.")
+					}
+				}
+				return err
+			}
+
+			if host.Platform == "windows" || host.Platform == "darwin" {
+				if host.MDM.EnrollmentStatus == nil || !strings.HasPrefix(*host.MDM.EnrollmentStatus, "On") ||
+					host.MDM.Name != fleet.WellKnownMDMFleet {
+					return errors.New(`Can't lock the host because it doesn't have MDM turned on.`)
+				}
+			}
+
+			if err := client.MDMLockHost(host.ID); err != nil {
+				return fmt.Errorf("Failed to lock host: %w", err)
+			}
+
+			fmt.Fprintf(c.App.Writer, `
+The host will lock when it comes online.  
+
+Copy and run this command to see lock status:
+
+fleetctl get host %s
+
+When you're ready to unlock the host, copy and run this command: 
+
+fleetctl mdm unlock --host=%s
+
+`, hostIdent, hostIdent)
```

Comment:
More as a FYI for a subtle feature of the `fmt` package than anything else (no need to change the PR), you can use `fleetctl mdm unlock --host=%[1]s` and pass the `hostIdent` only once to the list of arguments, and it will reuse the 1st argument, just in case you didn't know about it.